packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -169,6 +175,37 @@ export class QueuedRequestController extends BaseController<
       `${controllerName}:enqueueRequest`,
       this.enqueueRequest.bind(this),
     );
+
+    this.messagingSystem.subscribe(
+      SelectedNetworkControllerEventTypes.stateChange,
+      (_, patch) => {
+        patch.forEach(({ op, path }) => {
+          if (
```

Comment:
Nit: Hmm, I can see why you chose you use the patch here. I don't love seeing patches used for this because this code ends up being a bit fragile; different patch operations than expected would break this. This would also break depending on how domains were updated, e.g. if we did `domains = { 'foo.com': 'networkClientId', ...domains }` instead of `domains['foo.com'] = 'networkClientId'`, it might result in a `domains` replacement patch rather than a `domains['foo.com']` replacement patch. Some tricky assumptions here.

The obvious alternative approach would be to cache the entire `domains` state here and spot the difference, but that's a bit inelegant and inefficient compared to the patch approach, though still more robust.

Maybe we could add an event to the `SelectedNetworkController` that specifically tells us when a network is switched? That would make this much simpler.